ReactWindows/ReactNative/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,258 @@
+        /// </summary>
+        /// <param name="uiElementAutomationPeer"></param>
+        /// <returns></returns>
+        private static bool IsHiddenByAncestor(AutomationPeer uiElementAutomationPeer)
+        {
+            if (uiElementAutomationPeer == null)
+            {
+                return false;
+            }
+
+            bool result;
+            switch (uiElementAutomationPeer.Navigate(AutomationNavigationDirection.Parent))
+            {
+                case FrameworkElementAutomationPeer parentFEAutomationPeer:
+                    if (parentFEAutomationPeer.Owner != null)
+                    {
+                        var importantForAccessibilityAttached = GetImportantForAccessibilityAttached(parentFEAutomationPeer.Owner);
+                        if (importantForAccessibilityAttached == ImportantForAccessibility.Yes ||
+                            importantForAccessibilityAttached == ImportantForAccessibility.NoHideDescendants)
+                        {
+                            return true;
+                        }
+                    }
+                    result = IsHiddenByAncestor(parentFEAutomationPeer);
+                    break;
+                case ItemAutomationPeer parentIAutomationPeer:
+                    if (parentIAutomationPeer.Item is UIElement parentUIElement)
+                    {
+                        var importantForAccessibilityAttached = GetImportantForAccessibilityAttached(parentUIElement);
+                        if (importantForAccessibilityAttached == ImportantForAccessibility.Yes ||
+                            importantForAccessibilityAttached == ImportantForAccessibility.NoHideDescendants)
+                        {
+                            return true;
+                        }
+                    }
+                    result = IsHiddenByAncestor(parentIAutomationPeer);
+                    break;
+                case AutomationPeer parentAutomationPeer:
+                    result = IsHiddenByAncestor(parentAutomationPeer);
+                    break;
+                case null:
+                    result = false;
+                    break;
+                default:
+                    result = false;
+                    break;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Recursively sets the value of AccessibilityView according ImportantForAccessibility setting
+        /// for children of <paramref name="uiElementAutomationPeer"/>
+        /// </summary>
+        /// <param name="uiElementAutomationPeer"></param>
+        private static void SetChildrenAccessibilityViewFromImportantForAccessibility(AutomationPeer uiElementAutomationPeer)
+        {
+            if (uiElementAutomationPeer?.GetChildren() == null)
+            {
+                return;
+            }
+
+            foreach (AutomationPeer automationPeer in uiElementAutomationPeer.GetChildren())
```

Comment:
>automationPeer [](start = 36, length = 14)

Should we rename this to `childAutomationPeer`?

Reply:
Seems better readable, changed.

---
In reply to: [164943257](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r164943257) [](ancestors = 164943257)